### PR TITLE
[ALOY-1325] Add PreviewContext element to XML views

### PR DESCRIPTION
**Pull request by feons**
https://jira.appcelerator.org/browse/ALOY-1325

### CODE REVIEW

File: Alloy/commands/compile/parsers/default.js [hunk 1]
```diff
@@ -15,6 +15,10 @@ function parse(node, state, args) {
 		parts = fullname.split('.'),
 		extras = [];
 
+	if (node.previewConext) {
```
Comment (FokkeZB) [inline Alloy/commands/compile/parsers/default.js]: @feons typo? previewConext > previewContext

File: Alloy/commands/compile/parsers/default.js [hunk 2]
```diff
@@ -15,6 +15,10 @@ function parse(node, state, args) {
...
+		extras.push(['previewContext', node.previewConext]);
```
Comment (FokkeZB) [inline Alloy/commands/compile/parsers/default.js]: Same typo

File: Alloy/commands/compile/parsers/Ti.UI.iOS.PreviewContext.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+		state.extraStyle = styler.createVariableStyle(extraStyle);
+	}
+
+	previewContextState = require('./default').parse(node, state);
+	code += previewContextState.code;
+
+	if (parentViewId) {
+		code += parentViewId + '.previewContext = ' + previewContextState.parent.symbol + ';';
+	} else {
+		node.parentNode.previewConext = previewContextState.parent.symbol;
```
Comment (FokkeZB) [inline Alloy/commands/compile/parsers/Ti.UI.iOS.PreviewContext.js]: Same typo

**Comment by FokkeZB** [comment]
@feons one more thing in addition to the typo:

Could you add shorthands for previewAction and Group style? (`DESTRUCTIVE` > `Ti.UI.iOS.PREVIEW_ACTION_STYLE_DESTRUCTIVE` ?

Will this make the Alloy shipping with 5.1.0? It would be make the sample app a lot cleaner.

File: Alloy/commands/compile/parsers/default.js (hunk 1 above)
Comment (feons) [inline Alloy/commands/compile/parsers/default.js]: @FokkeZB, yup that's one series of epic type. Thank you for catching it!

File: Alloy/commands/compile/parsers/Ti.UI.iOS.PreviewAction.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+exports.parse = function(node, state) {
+	return require('./base').parse(node, state, parse);
+};
+
+function parse(node, state, args) {
+	var actionStyle = node.getAttribute('style');
+	if (actionStyle && actionStyle.indexOf('.UI.iOS.') === -1) {
+		node.setAttribute('style', 'Ti.UI.iOS.PREVIEW_ACTION_STYLE_.' + actionStyle.toUpperCase());
```
Comment (FokkeZB) [inline Alloy/commands/compile/parsers/Ti.UI.iOS.PreviewAction.js]: This renders to:

```
    $.__views.__alloyId44 = Ti.UI.iOS.createPreviewAction({
        title: "Send per Mail",
        style: Ti.UI.iOS.PREVIEW_ACTION_STYLE_.SELECTED,
        id: "__alloyId44"
    });
```

Remove the `.` after `STYLE_`

File: Alloy/commands/compile/parsers/Ti.UI.iOS.PreviewActionGroup.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+};
+
+function parse(node, state, args) {
+	var code = '',
+		groupState,
+		children = [],
+		actionStyle = node.getAttribute('style');
+
+	if (actionStyle && actionStyle.indexOf('.UI.iOS.') === -1) {
+		node.setAttribute('style', 'Ti.UI.iOS.PREVIEW_ACTION_STYLE_.' + actionStyle.toUpperCase());
```
Comment (FokkeZB) [inline Alloy/commands/compile/parsers/Ti.UI.iOS.PreviewActionGroup.js]: Same here: `_.` should be `_`

**Comment by FokkeZB** [comment]
Apart from the 2 new typos, verified fixed.

**Comment by feons** [comment]
@FokkeZB, thank you for catching the series of epic typo! **Approved.**

**Pull request merged**